hclsyntax/expression.go

```diff
@@ -678,6 +688,27 @@ func (e *functionCallDiagExtra) FunctionCallError() error {
 	return e.functionCallError
 }
 
+// FunctionCallUnknown is an interface implemented by a value in the Extra
+// field of some diagnostics to indicate when the error was caused by a call to
+// an unknown function.
+type FunctionCallUnknown interface {
```

Comment:
The existing "diagnostic extra" interface nearby here has "DiagExtra" at the end of its name to try to make it clear that it's there only for used in the "diagnostic extra" field. Should we do that here too?

(If we do end up finding another use of this interface that isn't related to "diagnostic extra" then we can presumably add another name covering the same methods to make it back-compatible, but I'm not imagining any specific reason why we might do that in future.)

Reply:
Yeah, that's a good point. I think I removed that at some point because this fell more natural to describe the error, but it does lose the context that it's only for he `Extra` field.